src/app/qgsidentifyresultsdialog.cpp

```diff
@@ -709,6 +712,10 @@ QgsIdentifyResultsFeatureItem *QgsIdentifyResultsDialog::createFeatureItem( QgsV
       defVal = vlayer->dataProvider()->defaultValueClause( fields.fieldOriginIndex( i ) );
 
     const QString originalValue = defVal == attrs.at( i ) ? defVal : fields.at( i ).displayString( attrs.at( i ) );
+    if ( originalValue.size() == 0 && QgsSettings().value( QStringLiteral( "/Map/hideNullValues" ), false ).toBool() )
```

Comment:
I don't think @3nids wants anymore raw QgsSettings use in the code anymore -- this should instead be ported to the new settings framework. See e.g. qgssettingsregistryapp.cpp